core/systems/performance/performance_manager.gd

```diff
@@ -610,31 +613,55 @@ func _read_amd_gpu_power_profile() -> void:
 
 ## Retrieves the current TDP from ryzenadj for AMD APU's.
 func _read_amd_tdp() -> void:
+	var set_sane_defaults: bool = false
+
 	var output: Array = await _async_do_exec(POWERTOOLS_PATH, ["ryzenadj", "-i"])
 	var exit_code := output[1] as int
 	if exit_code:
 		logger.info("Got exit code: " +str(exit_code) +". Unable to verify current tdp. Setting TDP to midpoint of range")
-		await _set_tdp_midpoint()
+		await _set_sane_defaults()
 		return
+
 	var result := (output[0][0] as String).split("\n")
 	var current_fastppt := 0.0
+
 	for setting in result:
 		var parts := setting.split("|")
-		var i := 0
-		for part in parts:
-			parts[i] = part.strip_edges()
-			i+=1
+		for index in parts.size():
+			parts[index] = parts[index].strip_edges()
+
 		if len(parts) < 3:
 			continue
-		match parts[1]:
-			"PPT LIMIT FAST":
-				current_fastppt = float(parts[2])
-			"STAPM LIMIT":
-				tdp_current = float(parts[2])
-			"THM LIMIT CORE":
-				gpu_temp_current = float(parts[2])
+
+		if parts[2] != "nan":
```

Comment:
Instead of explicitly checking for `nan`, what if you instead do something like:

```gdscript
var value := parts[2] as String
if value.is_valid_float():
    ...
```